packages/cli/test/shared/mocking.ts

```diff
@@ -1,11 +1,9 @@
 import { Container } from 'typedi';
 import { mock } from 'jest-mock-extended';
 import type { DeepPartial } from 'ts-essentials';
+import type { Class } from 'n8n-core';
 
-export const mockInstance = <T>(
-	ctor: new (...args: unknown[]) => T,
-	data: DeepPartial<T> | undefined = undefined,
-) => {
+export const mockInstance = <T>(ctor: Class<T>, data: DeepPartial<T> | undefined = undefined) => {
```

Comment:
1. Nit - for consistency, `cls` instead of `ctor`
2. I notice sometimes we use `DeepPartial` from `ts-essentials` and sometimes from `typeorm` - should we unify?

Reply:
1. done
2. i tried replace all `DeepPartial` with the one from `ts-essentials` a while ago (including our custom `DeepPartial` in the workflow package), but it threw a bunch of errors. so, it might be a bigger change than just changing the imports.